autorest/adal/token.go

```diff
@@ -913,6 +914,10 @@ func retryForIMDS(sender Sender, req *http.Request, maxAttempts int) (resp *http
 	delay := time.Duration(0)
 
 	for attempt < maxAttempts {
+		if resp != nil && resp.Body != nil {
```

Comment:
can we use `DrainResponseBody` here ?

Reply:
Unfortunately no as it would introduce a circular dependency.